src/validators/model.rs

```diff
@@ -250,9 +265,16 @@ impl ModelValidator {
 
     fn create_class(&self, model_dict: &PyAny, fields_set: &PyAny) -> PyResult<PyObject> {
         let py = model_dict.py();
-        let instance = create_class(self.class.as_ref(py))?;
-        set_model_attrs(instance.as_ref(py), model_dict, fields_set)?;
-        Ok(instance)
+        if self.custom_init {
+            let kwargs = PyDict::new(py);
+            let vd = ValidatedData::new(model_dict, fields_set);
+            kwargs.set_item(intern!(py, VALIDATED_DATA_KEY), vd.into_py(py))?;
+            self.class.call(py, (), Some(kwargs))
```

Comment:
I think it might work better if you do `BaseModel.call(py, (), Some(kwargs))` here, but I'm not sure, I'm not sure a good way to test it